src/utilities/sleep.hpp

```diff
@@ -0,0 +1,12 @@
+#ifndef SRC_UTILITIES_SLEEP_HPP_
+#define SRC_UTILITIES_SLEEP_HPP_
+
+#include "stm32g4xx_hal.h"
```

Comment:
why is it included in the header, but used in the implementation

Reply:
HAL_Delay is declared in this header file.